lib/function/arithmetic/log1p.js

```diff
@@ -0,0 +1,105 @@
+'use strict';
+
+var deepMap = require('../../utils/collection/deepMap');
+
+function factory (type, config, load, typed) {
+  var divideScalar = load(require('./divideScalar'));
+  var log = load(require('./log'));
+
+  /**
+   * Calculate the logarithm of a `value+1`.
+   *
+   * For matrices, the function is evaluated element wise.
+   *
+   * Syntax:
+   *
+   *    math.log1p(x)
+   *    math.log1p(x, base)
+   *
+   * Examples:
+   *
+   *    math.log1p(2.5);                 // returns 1.252762968495368
+   *    math.exp(math.log1p(1.4));       // returns 2.4
+   *
+   *    math.pow(10, 4);                 // returns 10000
+   *    math.log1p(9999, 10);            // returns 4
+   *    math.log1p(9999) / math.log(10); // returns 4
+   *
+   * See also:
+   *
+   *    exp, log, log2, log10
+   *
+   * @param {number | BigNumber | Complex | Array | Matrix} x
+   *            Value for which to calculate the logarithm of `x+1`.
+   * @param {number | BigNumber | Complex} [base=e]
+   *            Optional base for the logarithm. If not provided, the natural
+   *            logarithm of `x+1` is calculated.
+   * @return {number | BigNumber | Complex | Array | Matrix}
+   *            Returns the logarithm of `x+1`
+   */
+  var log1p = typed('log1p', {
+    'number': _log1pNumber,
+
+    'Complex': _log1pComplex,
+
+    'BigNumber': function (x) {
+      var y = x.plus(x.constructor.ONE);
+      if (!y.isNegative() || config.predictable) {
+        return y.ln();
```

Comment:
decimal.js does not have an implementation of log1p, I might open an issue there.